score_keeper_client/src/components/OldGame.jsx

```diff
@@ -1,37 +1,28 @@
 import React from 'react'
 import badminton from "../Pictures/pngegg.png";
+import { useEffect, useState } from 'react';
 
 function OldGame() {
 
-  // Get a reference to the element where you want to display the data
-  const resultsDiv = document.getElementById('results');
-
   // Make a GET request to your backend API
-  fetch('http://localhost:4000/getAllGameData')
-    .then(response => response.json())
-    .then(data => {
-      // Loop through the data and create a table row for each result
-      let resultsTable = '<table>';
-      data.forEach(result => {
-        resultsTable += `
-          <tr>
-            <td>${result.title}</td>
-            <td>${result.player1Name}</td>
-            <td>${result.player2Name}</td>
-            <td>${result.player1Score}</td>
-            <td>${result.player2Score}</td>
-            <td>${result.winText}</td>
-          </tr>
-        `;
-      });
-      resultsTable += '</table>';
-
-      // Display the results table on the webpage
-      resultsDiv.innerHTML = resultsTable;
-    })
-    .catch(error => console.error(error));
-
-
+  const [player1, setPlayer1] = useState('');
+  const [player2, setPlayer2] = useState('');
+  const [score1, setScore1] = useState(0);
+  const [score2, setScore2] = useState(0);
+  const [winText, setWinText] = useState('');
+
+  useEffect(() => {
+    fetch('http://localhost:4000/singleGameData/64145d0bb0a5c251d35886e5')
```

Comment:
Should not keep `http://localhost:4000` here, should use environment variable to store it. Later when you deploy your webapp, it becomes easier to replace the url and update it with domain name.